### PR TITLE
binary search problem

### PR DESCRIPTION
**Pull request by karitarr**
Problem: Search a sorted integer array for a value x and return whether or not x is in the array.
(*Using Iteration, will try using recursion and classes on another branch)
@mLewisLogic @ianlogan @joshualee @lxdvs

### CODE REVIEW

File: binarysearch.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+x = 2
+
+def binary_search(x, input_array)
+
+  #left and right are the start and end parameters of the array being evaluated at each loop
+  left = 0
+  right = input_array.length - 1
+
+  #evaluate whether x is greater than or less than the mid, then change the parameters accordingly
+  while left <= right
```
Comment (ianlogan) [inline binarysearch.rb]: Question for you @karitarr :

what advantage do you get by using iteration instead of recursion?

Comment (karitarr) [inline binarysearch.rb]: in this case- just practice solving it in a specific way :) but if you're asking in general, the real advantage in one over the other seems to be around how easy or hard it makes it to understand the algorithm. This one feels easier to follow recursively, but I wanted to make sure I could do it both ways.

Comment (ianlogan) [inline binarysearch.rb]: Cool. Note that recursion also takes up memory on the call stack, or more simply stack - take a look for more info here:

http://en.wikipedia.org/wiki/Call_stack

Comment (karitarr) [inline binarysearch.rb]: so that's where stackoverflow gets its name :)

**Comment by mLewisLogic** [comment]
:+1:

**Pull request merged**